### DISCUSSION TITLE
Approach circle is accelerating

### DISCUSSION THREAD

**Discussion by tiseppw** [ppy/osu#16515] [Q&A]
I don't know how to describe it, but...
Sometimes **(when map starting immediately)**, first note approach circle behaves strangely. Specifically:

- **Overlapping:**

https://user-images.githubusercontent.com/33760265/150023969-cab7276d-d257-4e54-b3bb-9508437e4bf7.mp4

- **Overlapping with shift:**

https://user-images.githubusercontent.com/33760265/150024007-204097c9-c6c0-4311-b5bb-c49826ed7cb6.mp4

I layered one note on another using montage to compare the speed.

This map: https://osu.ppy.sh/beatmapsets/339091#osu/750244
Other example: https://osu.ppy.sh/beatmapsets/1256111#osu/2610417

This bug is not a serious problem, but it becomes very annoying if the map is fast or starts without music.

P.S. Hope I didn't duplicate the question.

**Comment by tiseppw** [comment]
I think adding a second before the start of the map would fix the bug.

**Reply by bdach** [reply]
sorry but no. that sounds like a workaround rather than an actual fix.

**Reply by tiseppw** [reply]
Ok, sure

**Comment by peppy** [comment]
@VladDen4 how are you recording the videos? they seem to show motion blur between frames. are you using a camera?

If possible could you do a proper screen capture?

**Reply by tiseppw** [reply]
I missed the notice. Sorry.
For recording I use OBS (60fps, ~9500 bits)
Monitor 60Hz
In-game fps 2x refresh rate (120fps)

**Comment by Pallid3** [comment]
I found this bug annoying as well and made video of my best try explaining it:
https://www.youtube.com/watch?v=FNnIlO2X9sE

**Reply by peppy** [reply]
This happens due to your audio device taking a moment to start playback of the audio track. Since we use the audio track as the master clock for gameplay, this can cause a slight pause when the track starts very close to the first hit circle (which is why you only notice it in some beatmaps).

There is a hope that we can better "cache" the start of playback at a BASS level to avoid this.
